### ISSUE TITLE
`*args` positional parameter dropped out and became a phantom when using `Concatenate` to shorten parameters

### ISSUE THREAD

**Issue by NCBM** [microsoft/pyright#9630]
**Describe the bug**

When using `typing.Concatenate` to reduce parameters (just like a part of partial function), I found the positional parameter `*args` (or any other name) dropped out of the parameter list after a `Concatenate` operation and become a phantom in pyright, which is not printed in parameter type while type checking of the `*args` still works.

I also tested the code in pylance but the error is even not reported.

**Code or Screenshots**

```python
from typing import Callable, Concatenate, reveal_type


def _onepartial[T, **P, R](fn: Callable[Concatenate[T, P], R], val: T) -> Callable[P, R]:
    ...


def test2(p1: int, *ps: str) -> None:
    ...


reveal_type(_onepartial(test2, 3))
reveal_type(_onepartial(_onepartial(test2, 3), "foo"))  # *ps dropped out
reveal_type(
    _onepartial(
        _onepartial(
            _onepartial(test2, 3),
            "foo"
        ),
        "bar"  # no error since *ps type match, although which becomes a phantom
    )
)
reveal_type(
    _onepartial(
        _onepartial(
            _onepartial(test2, 3),
            "foo"
        ),
        3  # error since *ps type mismatch, although which becomes a phantom
    )
)
```

[View in pyright-play.net](https://pyright-play.net/?strict=true&locale=en-us&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoDCAhgDYlEBGJApgDSFgoDGRM1KrdUI1AbtaQD68BNQBQEgCbVgUQY2oIieJKQDaAFXoAqbQAV6AJQC6ACmAoAXIVLkq1NQUYs2HNpvp7jR71F6lrDQBKKABaAD4bMkoaNQMoE0sxKBSoADoMiTEAYihpWTYAZxhTBABGa2KQegQAJmtUGBqAZkqYEBCIqAA5BSTc1PTMqRlYamLa0orMFCaobQRCto6wyN6UaiTBjLSs3NyYAAtqOCgAd1wAa0Kc7j4BEmFEalN5DaUVUiDbnn4hEReb0UyhgqhIpiKcwARMAwGAoUFvrlfg8nqJXgoPqDSBj3iCwRDxtDYfCgvRmoiJCj-s9ccDPuDIbVyZTqY8AXSsQSgVycUyWfQYXCESEoLkFoU8uAEKJJFAwABXGBiNlol7JVI8-E4jWDFJahmmXV6wYG7GMonMqAU2jGk0pIXwu2pMnOh0UZRQlK5FBgKDUEDgEBQQqoJjUeaLWDPKAQVhMQ70UhHRVoQ7nQ5IBNQCjUJiQcZQIhQBCHIizSDG77fVUc41mgluuSY7XgpumluG-nW132vWOqFN3v25re-2B3AhsMRiXR0SxpCFOMwBNJkgphVpjNZ9O5-MQQvF0vlmCVwbVsRAA)

**VS Code extension or command-line**

Tested in:

- pyright-play.net: pyright 1.1.391, python 3.13;
- vscode: pylance 2024.12.1, python 3.13, however not reporting the error as commented.

**Comment by erictraut** [comment]
Thanks for the issue. The typing spec isn't entirely clear how a `Concatenate` should be used in conjunction with a `*args` parameter, but I agree that your interpretation (which is consistent with mypy's interpretation) makes sense. I'll change pyright's behavior to match. This will be included in the next release.

Pylance doesn't show this error because it defaults `typeCheckingMode` to "off". If you set this to "standard", you'll see the same errors reported by pyright.

**Comment by NCBM** [comment]
thanks. maybe I forgot to change the `typeCheckingMode`.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.392.
